### ISSUE TITLE
Pytorch docker build fail

### ISSUE THREAD

**Issue by krumo** [pytorch/pytorch#10687]
## Issue description

When I am trying to build a Pytorch docker container, I encounter the following errors:

```
    Install the project...
    -- Install configuration: "Release"
    -- Installing: /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/lib/libTHD.a
    -- Installing: /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THD/THD.h
    -- Installing: /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THD/base/THDGenerateAllTypes.h
    -- Installing: /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THD/base/TensorDescriptor.h
    -- Installing: /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THD/base/Cuda.h
    -- Installing: /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THD/base/ChannelType.h
    -- Installing: /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THD/base/DataChannelRequest.h
    -- Installing: /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THD/base/DataChannel.h
    -- Installing: /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THD/process_group/Collectives.h
    -- Installing: /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THD/process_group/General.h
    + popd
    /tmp/pip-req-build-2s8ksvn3/torch/lib /tmp/pip-req-build-2s8ksvn3/build
    ++ uname
    + [[ Linux == \D\a\r\w\i\n ]]
    + popd
    /tmp/pip-req-build-2s8ksvn3/build
    + for arg in '"$@"'
    + [[ c10d == \n\c\c\l ]]
    + [[ c10d == \g\l\o\o ]]
    + [[ c10d == \c\a\f\f\e\2 ]]
    + [[ c10d == \T\H\D ]]
    + [[ c10d == \l\i\b\s\h\m ]]
    + [[ c10d == \l\i\b\s\h\m\_\w\i\n\d\o\w\s ]]
    + [[ c10d == \c\1\0\d ]]
    + pushd /tmp/pip-req-build-2s8ksvn3/torch/lib
    /tmp/pip-req-build-2s8ksvn3/torch/lib /tmp/pip-req-build-2s8ksvn3/build
    + build c10d
    + [[ -z '' ]]
    + CMAKE_ARGS=()
    + mkdir -p build/c10d
    + pushd build/c10d
    /tmp/pip-req-build-2s8ksvn3/torch/lib/build/c10d /tmp/pip-req-build-2s8ksvn3/torch/lib /tmp/pip-req-build-2s8ksvn3/build
    + BUILD_C_FLAGS=
    + case $1 in
    + BUILD_C_FLAGS=' -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include"   -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/TH" -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THC"   -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THS" -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THCS"   -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THNN" -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THCUNN" -DOMPI_SKIP_MPICXX=1 -fexceptions'
    + cmake ../../c10d -DCMAKE_MODULE_PATH=/tmp/pip-req-build-2s8ksvn3/cmake/Modules_CUDA_fix -DTorch_FOUND=1 -DCMAKE_INSTALL_PREFIX=/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install '-DCMAKE_C_FLAGS= -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include"   -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/TH" -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THC"   -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THS" -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THCS"   -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THNN" -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THCUNN" -DOMPI_SKIP_MPICXX=1 -fexceptions ' '-DCMAKE_CXX_FLAGS= -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include"   -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/TH" -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THC"   -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THS" -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THCS"   -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THNN" -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THCUNN" -DOMPI_SKIP_MPICXX=1 -fexceptions  -std=c++11  ' '-DCMAKE_EXE_LINKER_FLAGS=-L"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/lib"  -Wl,-rpath,$ORIGIN ' '-DCMAKE_SHARED_LINKER_FLAGS=-L"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/lib"  -Wl,-rpath,$ORIGIN ' -DCMAKE_INSTALL_LIBDIR=/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/lib '-DCUDA_NVCC_FLAGS= -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include"   -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/TH" -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THC"   -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THS" -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THCS"   -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THNN" -I"/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/THCUNN" -DOMPI_SKIP_MPICXX=1' -DCUDA_DEVICE_DEBUG=0 -DCMAKE_PREFIX_PATH=/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install '-Dcwrap_files=/tmp/pip-req-build-2s8ksvn3/torch/lib/ATen/Declarations.cwrap;/tmp/pip-req-build-2s8ksvn3/torch/lib/THNN/generic/THNN.h;/tmp/pip-req-build-2s8ksvn3/torch/lib/THCUNN/generic/THCUNN.h;/tmp/pip-req-build-2s8ksvn3/torch/lib/ATen/nn.yaml' -DTH_INCLUDE_PATH=/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include -DTH_LIB_PATH=/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/lib -DTH_LIBRARIES=/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/lib/libTH.so -DCAFFE2_LIBRARIES=/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/lib/libcaffe2.so -DTHNN_LIBRARIES=/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/lib/libTHNN.so -DTHCUNN_LIBRARIES=/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/lib/libTHCUNN.so -DTHS_LIBRARIES=/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/lib/libTHS.so -DTHC_LIBRARIES=/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/lib/libTHC.so -DTHCS_LIBRARIES=/tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/lib/libTHCS.so -DTH_SO_VERSION=1 -DTHC_SO_VERSION=1 -DTHNN_SO_VERSION=1 -DTHCUNN_SO_VERSION=1 -DTHD_SO_VERSION=1 -DUSE_CUDA=1 -DNO_NNPACK=0 -DNCCL_EXTERNAL=1 -Dnanopb_BUILD_GENERATOR=0 -DCMAKE_DEBUG_POSTFIX= -DCMAKE_BUILD_TYPE=Release -DCMAKE_EXPORT_COMPILE_COMMANDS=1
    -- The C compiler identification is GNU 5.4.0
    -- The CXX compiler identification is GNU 5.4.0
    -- Check for working C compiler: /usr/bin/cc
    -- Check for working C compiler: /usr/bin/cc -- works
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- Check for working CXX compiler: /usr/bin/c++
    -- Check for working CXX compiler: /usr/bin/c++ -- works
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- Looking for pthread.h
    -- Looking for pthread.h - found
    -- Looking for pthread_create
    -- Looking for pthread_create - not found
    -- Looking for pthread_create in pthreads
    -- Looking for pthread_create in pthreads - not found
    -- Looking for pthread_create in pthread
    -- Looking for pthread_create in pthread - found
    -- Found Threads: TRUE
    -- Caffe2: Found protobuf with new-style protobuf targets.
    -- Caffe2: Protobuf version 3.5.0
    -- Found CUDA: /usr/local/cuda (found suitable version "9.0", minimum required is "7.0")
    -- Caffe2: CUDA detected: 9.0
    -- Caffe2: CUDA nvcc is: /usr/local/cuda/bin/nvcc
    -- Caffe2: CUDA toolkit directory: /usr/local/cuda
    -- Caffe2: Header version is: 9.0
    -- Found CUDNN: /usr/include/
    -- Found cuDNN: v7.1.4  (include: /usr/include/, library: /usr/lib/x86_64-linux-gnu/libcudnn.so.7)
    CMake Warning at /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/share/cmake/Caffe2/public/utils.cmake:173 (message):
      In the future we will require one to explicitly pass TORCH_CUDA_ARCH_LIST
      to cmake instead of implicitly setting it as an env variable.  This will
      become a FATAL_ERROR in future version of pytorch.
    Call Stack (most recent call first):
      /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/share/cmake/Caffe2/public/cuda.cmake:332 (torch_cuda_get_nvcc_gencode_flag)
      /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/share/cmake/Caffe2/Caffe2Config.cmake:79 (include)
      CMakeLists.txt:13 (find_package)


    -- Added CUDA NVCC flags for: -gencode;arch=compute_35,code=sm_35;-gencode;arch=compute_52,code=sm_52;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_70,code=compute_70
    -- Found Gloo: /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include
    -- Gloo_LIBRARY: /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/lib/libgloo_cuda.a
    -- Gloo_NATIVE_LIBRARY: /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/lib/libgloo.a
    -- Gloo_INCLUDE_DIR: /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include
    -- Could NOT find MPI_C (missing:  MPI_C_LIBRARIES MPI_C_INCLUDE_PATH)
    -- Could NOT find MPI_CXX (missing:  MPI_CXX_LIBRARIES MPI_CXX_INCLUDE_PATH)
    -- Not able to find MPI, will compile c10d without MPI support
    -- Found NCCL: /usr/include
    -- Determining NCCL version from the header file: /usr/include/nccl.h
    -- NCCL_MAJOR_VERSION: 2
    -- Found NCCL (include: /usr/include, library: /usr/lib/x86_64-linux-gnu/libnccl.so)
    -- NCCL_LIBRARIES: /usr/lib/x86_64-linux-gnu/libnccl.so
    -- NCCL_INCLUDE_DIRS: /usr/include
    -- NCCL Version 2 or higher found, will compile with NCCL distributed backend
    -- Found CUDA: /usr/local/cuda (found version "9.0")
    -- Configuring done
    -- Generating done
    CMake Warning:
      Manually-specified variables were not used by the project:

        CAFFE2_LIBRARIES
        CMAKE_DEBUG_POSTFIX
        CMAKE_INSTALL_LIBDIR
        NCCL_EXTERNAL
        NO_NNPACK
        THCS_LIBRARIES
        THCUNN_LIBRARIES
        THCUNN_SO_VERSION
        THC_LIBRARIES
        THC_SO_VERSION
        THD_SO_VERSION
        THNN_LIBRARIES
        THNN_SO_VERSION
        THS_LIBRARIES
        TH_INCLUDE_PATH
        TH_LIBRARIES
        TH_LIB_PATH
        TH_SO_VERSION
        Torch_FOUND
        USE_CUDA
        cwrap_files
        nanopb_BUILD_GENERATOR


    -- Build files have been written to: /tmp/pip-req-build-2s8ksvn3/torch/lib/build/c10d
    + make install -j80
    Scanning dependencies of target c10d
    [  4%] Building CXX object CMakeFiles/c10d.dir/CUDAUtils.cpp.o
    [  8%] Building CXX object CMakeFiles/c10d.dir/FileStore.cpp.o
    [ 13%] Building CXX object CMakeFiles/c10d.dir/Store.cpp.o
    [ 17%] Building CXX object CMakeFiles/c10d.dir/TCPStore.cpp.o
    [ 21%] Building CXX object CMakeFiles/c10d.dir/ProcessGroup.cpp.o
    [ 26%] Building CXX object CMakeFiles/c10d.dir/ProcessGroupGloo.cpp.o
    [ 30%] Building CXX object CMakeFiles/c10d.dir/Utils.cpp.o
    [ 34%] Building CXX object CMakeFiles/c10d.dir/ProcessGroupNCCL.cpp.o
    /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/ProcessGroupGloo.cpp: In member function 'c10d::AlgorithmEntry* c10d::ProcessGroupGloo::checkout(const c10d::AlgorithmKey&)':
    /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/ProcessGroupGloo.cpp:445:18: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       if (vec.size() != cacheNumAlgorithmEntries_) {
                      ^
    [ 39%] Linking CXX static library libc10d.a
    [ 39%] Built target c10d
    [ 43%] Building NVCC (Device) object test/CMakeFiles/c10d_cuda_test.dir/c10d_cuda_test_generated_CUDATest.cu.o
    Scanning dependencies of target FileStoreTest
    Scanning dependencies of target TCPStoreTest
    [ 47%] Building CXX object test/CMakeFiles/FileStoreTest.dir/FileStoreTest.cpp.o
    [ 52%] Building CXX object test/CMakeFiles/TCPStoreTest.dir/TCPStoreTest.cpp.o
    Scanning dependencies of target allreduce
    [ 56%] Building CXX object example/CMakeFiles/allreduce.dir/allreduce.cpp.o
    [ 60%] Linking CXX executable FileStoreTest
    [ 60%] Built target FileStoreTest
    /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/example/allreduce.cpp: In function 'int main(int, char**)':
    /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/example/allreduce.cpp:16:64: warning: 'at::Tensor at::ones(const at::Type&, at::IntList)' is deprecated [-Wdeprecated-declarations]
         auto x = at::ones(at::CPU(at::kFloat), {1000, 16 * (i + 1)});
                                                                    ^
    In file included from /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/ATen.h:13:0,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/../c10d/ProcessGroup.hpp:7,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/../c10d/ProcessGroupGloo.hpp:19,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/example/allreduce.cpp:2:
    /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/Functions.h:3949:22: note: declared here
     static inline Tensor ones(const Type & dtype, IntList size) {
                          ^
    [ 65%] Linking CXX executable TCPStoreTest
    [ 69%] Linking CXX executable allreduce
    [ 69%] Built target TCPStoreTest
    [ 69%] Built target allreduce
    Scanning dependencies of target c10d_cuda_test
    [ 73%] Linking CXX static library libc10d_cuda_test.a
    [ 73%] Built target c10d_cuda_test
    Scanning dependencies of target ProcessGroupGlooTest
    Scanning dependencies of target ProcessGroupNCCLTest
    Scanning dependencies of target ProcessGroupGlooAsyncTest
    [ 78%] Building CXX object test/CMakeFiles/ProcessGroupGlooTest.dir/ProcessGroupGlooTest.cpp.o
    [ 82%] Building CXX object test/CMakeFiles/ProcessGroupNCCLTest.dir/ProcessGroupNCCLTest.cpp.o
    [ 86%] Building CXX object test/CMakeFiles/ProcessGroupGlooAsyncTest.dir/ProcessGroupGlooAsyncTest.cpp.o
    /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupNCCLTest.cpp: In member function 'std::vector<at::Tensor> NCCLTest::getTensors()':
    /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupNCCLTest.cpp:99:49: error: no matching function for call to 'at::Tensor::toBackend(const at::DeviceType&)'
           outputs[i] = inputs_[i].toBackend(at::kCPU);
                                                     ^
    In file included from /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/NativeFunctions.h:7:0,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/Functions.h:11,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/ATen.h:13,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/../c10d/ProcessGroup.hpp:7,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/../c10d/ProcessGroupNCCL.hpp:8,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupNCCLTest.cpp:5:
    /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/TensorMethods.h:38:15: note: candidate: at::Tensor at::Tensor::toBackend(at::Backend) const
     inline Tensor Tensor::toBackend(Backend b) const {
                   ^
    /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/TensorMethods.h:38:15: note:   no known conversion for argument 1 from 'const at::DeviceType' to 'at::Backend'
    /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupNCCLTest.cpp: In member function 'std::vector<std::vector<at::Tensor> > NCCLTest::getOutputTensors()':
    /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupNCCLTest.cpp:118:58: error: no matching function for call to 'at::Tensor::toBackend(const at::DeviceType&)'
             outputs[i][j] = outputs_[i][j].toBackend(at::kCPU);
                                                              ^
    In file included from /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/NativeFunctions.h:7:0,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/Functions.h:11,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/ATen.h:13,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/../c10d/ProcessGroup.hpp:7,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/../c10d/ProcessGroupNCCL.hpp:8,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupNCCLTest.cpp:5:
    /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/TensorMethods.h:38:15: note: candidate: at::Tensor at::Tensor::toBackend(at::Backend) const
     inline Tensor Tensor::toBackend(Backend b) const {
                   ^
    /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/TensorMethods.h:38:15: note:   no known conversion for argument 1 from 'const at::DeviceType' to 'at::Backend'
    /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupGlooTest.cpp: In member function 'std::shared_ptr<c10d::ProcessGroup::Work> SignalTest::run(int, int)':
    /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupGlooTest.cpp:51:47: warning: 'at::Tensor at::ones(const at::Type&, at::IntList)' is deprecated [-Wdeprecated-declarations]
             at::ones(at::CPU(at::kFloat), {16, 16}),
                                                   ^
    In file included from /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/ATen.h:13:0,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/../c10d/ProcessGroup.hpp:7,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/../c10d/ProcessGroupGloo.hpp:19,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupGlooTest.cpp:16:
    /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/Functions.h:3949:22: note: declared here
     static inline Tensor ones(const Type & dtype, IntList size) {
                          ^
    /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupGlooTest.cpp: In function 'void testAllreduce(const string&, at::Backend)':
    /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupGlooTest.cpp:158:64: warning: 'at::Tensor at::ones(const at::Type&, at::IntList)' is deprecated [-Wdeprecated-declarations]
         auto tensor = at::ones(at::getType(b, at::kFloat), {16, 16}) * i;
                                                                    ^
    In file included from /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/ATen.h:13:0,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/../c10d/ProcessGroup.hpp:7,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/../c10d/ProcessGroupGloo.hpp:19,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupGlooTest.cpp:16:
    /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/Functions.h:3949:22: note: declared here
     static inline Tensor ones(const Type & dtype, IntList size) {
                          ^
    /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupGlooTest.cpp: In function 'void testBroadcast(const string&, at::Backend)':
    /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupGlooTest.cpp:208:49: warning: 'at::Tensor at::ones(const at::Type&, at::IntList)' is deprecated [-Wdeprecated-declarations]
               inputs[k][l] = at::ones(type, {16, 16}) * (k * stride + l);
                                                     ^
    In file included from /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/ATen.h:13:0,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/../c10d/ProcessGroup.hpp:7,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/../c10d/ProcessGroupGloo.hpp:19,
                     from /tmp/pip-req-build-2s8ksvn3/torch/lib/c10d/test/ProcessGroupGlooTest.cpp:16:
    /tmp/pip-req-build-2s8ksvn3/torch/lib/tmp_install/include/ATen/Functions.h:3949:22: note: declared here
     static inline Tensor ones(const Type & dtype, IntList size) {
                          ^
    test/CMakeFiles/ProcessGroupNCCLTest.dir/build.make:62: recipe for target 'test/CMakeFiles/ProcessGroupNCCLTest.dir/ProcessGroupNCCLTest.cpp.o' failed
    make[2]: *** [test/CMakeFiles/ProcessGroupNCCLTest.dir/ProcessGroupNCCLTest.cpp.o] Error 1
    CMakeFiles/Makefile2:375: recipe for target 'test/CMakeFiles/ProcessGroupNCCLTest.dir/all' failed
    make[1]: *** [test/CMakeFiles/ProcessGroupNCCLTest.dir/all] Error 2
    make[1]: *** Waiting for unfinished jobs....
    [ 91%] Linking CXX executable ProcessGroupGlooTest
    [ 95%] Linking CXX executable ProcessGroupGlooAsyncTest
    [ 95%] Built target ProcessGroupGlooTest
    [ 95%] Built target ProcessGroupGlooAsyncTest
    Makefile:138: recipe for target 'all' failed
    make: *** [all] Error 2
    Failed to run 'bash ../tools/build_pytorch_libs.sh --use-cuda --use-nnpack caffe2 nanopb libshm gloo THD c10d'
    Running setup.py install for torch: finished with status 'error'
Cleaning up...
  Removing source in /tmp/pip-req-build-2s8ksvn3
Command "/opt/conda/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-req-build-2s8ksvn3/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-ppouch5h/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-req-build-2s8ksvn3/
Exception information:
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/pip/_internal/basecommand.py", line 228, in main
    status = self.run(options, args)
  File "/opt/conda/lib/python3.6/site-packages/pip/_internal/commands/install.py", line 335, in run
    use_user_site=options.use_user_site,
  File "/opt/conda/lib/python3.6/site-packages/pip/_internal/req/__init__.py", line 49, in install_given_reqs
    **kwargs
  File "/opt/conda/lib/python3.6/site-packages/pip/_internal/req/req_install.py", line 779, in install
    spinner=spinner,
  File "/opt/conda/lib/python3.6/site-packages/pip/_internal/utils/misc.py", line 698, in call_subprocess
    % (command_desc, proc.returncode, cwd))
pip._internal.exceptions.InstallationError: Command "/opt/conda/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-req-build-2s8ksvn3/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-ppouch5h/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-req-build-2s8ksvn3/

```

## System Info

- PyTorch or Caffe2: PyTorch
- How you installed PyTorch (conda, pip, source): Docker
- Build command you used (if compiling from source):
- OS: Ubuntu 16.04
- PyTorch version: 0.4.1
- Python version: 3.6.6
- CUDA/cuDNN version: CUDA9.0, cuDNN 7
- GPU models and configuration: Pascal P100
- GCC version (if compiling from source): 5.4.0
- CMake version: 3.5.1

**Comment by krumo** [comment]
@zou3519 I build the image by running `nvidia-docker build --rm -t myname/pytorch -f pytorch/docker/pytorch/Dockerfile .`  under the Pytorch folder.

**Comment by mcarilli** [comment]
@zou3519 I just encountered this error as well.  I believe it is because PR [10478](https://github.com/pytorch/pytorch/commit/6bdbad93b96ecd943fe5091725ccd70fad9a6057) ("Refactor Device to not depend on Backend.") overlooked [ProcessGroupNCCLTest.cpp](https://github.com/pytorch/pytorch/commits/master/torch/lib/c10d/test/ProcessGroupNCCLTest.cpp).

**Comment by pietern** [comment]
This was fixed in #10704.

**Comment by fatalfeel** [comment]
~/pytorch/torch/lib/c10d/example/allreduce.cpp
//auto store = std::make_shared<FileStore>("/tmp/c10d_example", size);
//to
auto store = c10::make_intrusive<FileStore>("/tmp/c10d_example", size);

~done~
